docs/userguide/utils.md

```diff
@@ -0,0 +1,135 @@
+# both experimental and computational representation of the search space.
+parameters = [
+    NumericalDiscreteParameter(name=f"p{k+1}", values=np.linspace(0, 100, 20))
+    for k in range(10)
+]
+
+# Estimate the required memory for such a space
+mem_estimate = SearchSpace.estimate_product_space_size(parameters)
+
+# Print quantities of interest
+print("Experimental Representation")
+print(f"Estimated size: {mem_estimate.exp_rep_human_readable}")
+print(f"Estimated size in Bytes: {mem_estimate.exp_rep_bytes}")
+print(f"Expected data frame shape: {mem_estimate.exp_rep_shape}")
+
+print("Computational Representation")
+print(f"Estimated size: {mem_estimate.comp_rep_human_readable}")
+print(f"Estimated size in Bytes: {mem_estimate.comp_rep_bytes}")
+print(f"Expected data frame shape: {mem_estimate.comp_rep_shape}")
+```
+
+```{admonition} Estimation with Constraints
+:class: warning
+{meth}`~baybe.searchspace.core.SearchSpace.estimate_product_space_size`
+currently does not include the influence of potential constraints in your search space
+as it is generally very hard to incorporate the effect of arbitrary constraints without
+actually building the entire space. Hence, you should always **treat the number you get
+as upper bound** of required memory. This can still be useful – for instance if your
+estimate already is several Exabytes, it is unlikely that most computers would be able
+to handle the result even if there are constraints present.
+```
+
+```{admonition} Memory During Optimization
+:class: warning
+{meth}`~baybe.searchspace.core.SearchSpace.estimate_product_space_size`
+only estimates the memory required to handle the search space. **It does not estimate
+the memory required during optimization**, which can be of a similar magnitude, but
+generally depends on additional factors.
+```
+
+```{admonition} Influence of Continuous Parameters
+:class: info
+Continuous parameters do not influence the size of the discrete search space part.
+Hence, they are ignored by the utility.
+```
+
+```{admonition} Efficient Search Space Creation
+:class: tip
+If you run into issues creating large search spaces, as for instance in mixture
+use cases, you should consider resorting to more specialized ways of creation by
+invoking alternative search space constructors like 
+{meth}`~baybe.searchspace.discrete.SubspaceDiscrete.from_dataframe`
+or 
+{meth}`~baybe.searchspace.discrete.SubspaceDiscrete.from_simplex`.
+Instead of creating a product space first and then filtering it down
+according to constraints, they offer a more direct and thus efficient path to the 
+desired result, typically requiring substantially less memory. 
+For example, {meth}`~baybe.searchspace.discrete.SubspaceDiscrete.from_simplex` 
+includes the mixture constraint already *during* the product creation. 
+In addition, BayBE can also be installed with its optional `polars` dependency 
+(`pip install baybe[polars]`) that activates efficient machinery for constraint handling.
```

Comment:
Polars cannot be installed in computers with chips that do not have AVX support, see https://docs.pola.rs/user-guide/installation/#legacy-cpu. Since the M1 Chip is one of the chips affected by this, we might want to add a comment here that it might be necessary to manually install the "legacy" package. Not sure if this is the correct place for this information, however.

Reply:
please mark this to known issues or something, i dont think this admonition (which is essentially just a sidenote) is the place for this detail

Reply:
@AVHopp, can you create an issue that reminds us of adding it to known_issues?